userservice/src/main/java/ro/devdepot/services/implementation/UserServiceImpl.java

```diff
@@ -27,4 +33,42 @@ public LoginResponse login(LoginRequest loginRequest) {
 
         return null;
     }
+
+    @Override
+    public User createUser(User user) {
+        return userRepository.save(user);
+    }
+
+    @Override
+    public User updateUser(User user, Long id) {
+        User existingUser = userRepository.findById(id).orElse(null);
+
+        existingUser.setUsername(user.getUsername());
+
+        return userRepository.save(existingUser);
+    }
+
+    @Override
+    public User updateUserRole(UserRole newuserRole, Long id) {
+        User existingUser = userRepository.findById(id).orElse(null);
+
+        existingUser.setUserRole(newuserRole);
+
+        return userRepository.save(existingUser);
+    }
+
+    @Override
+    public User getUserById(Long id) {
+        return userRepository.findById(id).orElse(null);
+    }
+
+    @Override
+    public List<User> getAllUsers() {
+        return userRepository.findAll();
+    }
+
+    @Override
+    public void deleteUserById(Long id) {
+        userRepository.deleteById(id);
```

Comment:
should have validations: 
1. user with id exists (tip: create boolean method existsUserById in user repository)
2. the person who calls this is the logged in user or an admin so nobody else can delete other peoples accounts (comes after the security implementation, for now just add validation 1)